src/subcommand/wallet/resume.rs

```diff
@@ -27,8 +27,13 @@ impl Resume {
           continue;
         };
 
-        if wallet.is_mature(rune, &entry.commit)? {
-          etchings.push(wallet.send_etching(rune, &entry)?);
+        match wallet.check_maturity(*rune, &entry.commit)? {
+          Maturity::Mature => etchings.push(wallet.send_etching(*rune, entry)?),
+          Maturity::CommitSpent(txid) => {
+            eprintln!("Commitment for rune etching {rune} spent in {txid}");
+            etchings.remove(i);
+          }
+          _ => continue,
```

Comment:
I think it would be nice to explicitly enumerate all the cases, instead of using `_`, so that it's obvious that it's okay to ignore the other cases.

Reply:
makes sense, now that I have added them, just realized that we should also clear etching if commit is not found (for CommitNotFound arm)?

Reply:
done.